core-api/src/main/java/com/optimizely/ab/event/BatchEventProcessor.java

```diff
@@ -122,26 +123,46 @@ public void flush() throws InterruptedException {
         eventQueue.put(FLUSH_SIGNAL);
     }
 
+    private interface QueueService {
+        Object get() throws InterruptedException;
+    }
+
     public class EventConsumer implements Runnable {
         private LinkedList<UserEvent> currentBatch = new LinkedList<>();
         private long deadline = System.currentTimeMillis() + flushInterval;
 
         @Override
         public void run() {
             try {
+                int waitCount = 0;
+
+                QueueService polling = () -> eventQueue.poll(flushInterval, TimeUnit.MILLISECONDS);
```

Comment:
When polling the queue, should we block until the next deadline, instead of blocking for `flushInterval`?